it/google-cloud-platform/src/test/java/com/google/cloud/teleport/it/gcp/bigtable/BigTableIOLT.java

```diff
@@ -0,0 +1,256 @@
+
+  @Before
+  public void setup() throws IOException {
+    resourceManager = BigtableResourceManager.builder(testName, project).build();
+
+    String testConfig =
+        TestProperties.getProperty("configuration", "small", TestProperties.Type.PROPERTY);
+    configuration = TEST_CONFIGS.get(testConfig);
+    if (configuration == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unknown test configuration: [%s]. Known configs: %s",
+              testConfig, TEST_CONFIGS.keySet()));
+    }
+    // tempLocation needs to be set for bigtable IO writes
+    if (!Strings.isNullOrEmpty(tempBucketName)) {
+      String tempLocation = String.format("gs://%s/temp/", tempBucketName);
+      writePipeline.getOptions().as(TestPipelineOptions.class).setTempRoot(tempLocation);
+      writePipeline.getOptions().setTempLocation(tempLocation);
+      readPipeline.getOptions().as(TestPipelineOptions.class).setTempRoot(tempLocation);
+      readPipeline.getOptions().setTempLocation(tempLocation);
+    }
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(resourceManager);
+  }
+
+  private static final Map<String, Configuration> TEST_CONFIGS =
+      ImmutableMap.of(
+          "local", Configuration.of(1_000L, 5, "DirectRunner", 1000), // 1MB
+          "small", Configuration.of(1_000_000L, 20, "DataflowRunner", 1000), // 1 GB
+          "medium", Configuration.of(10_000_000L, 40, "DataflowRunner", 1000), // 10 GB
+          "large", Configuration.of(100_000_000L, 100, "DataflowRunner", 1000) // 100 GB
+          );
+
+  /** Run integration test with configurations specified by TestProperties. */
+  @Test
+  public void testWriteAndRead() throws IOException {
+    tableId = generateTableId(testName);
+    resourceManager.createTable(
+        tableId,
+        ImmutableList.of(COLUMN_FAMILY_NAME),
+        org.threeten.bp.Duration.ofMinutes(TABLE_MAX_AGE_MINUTES));
+
+    PipelineLauncher.LaunchInfo writeInfo = testWrite();
+    PipelineOperator.Result writeResult =
+        pipelineOperator.waitUntilDone(
+            createConfig(writeInfo, Duration.ofMinutes(configuration.getPipelineTimeout())));
+    assertNotEquals(PipelineOperator.Result.LAUNCH_FAILED, writeResult);
+
+    PipelineLauncher.LaunchInfo readInfo = testRead();
+    PipelineOperator.Result result =
+        pipelineOperator.waitUntilDone(
+            createConfig(readInfo, Duration.ofMinutes(configuration.getPipelineTimeout())));
+    assertNotEquals(PipelineOperator.Result.LAUNCH_FAILED, result);
+    assertEquals(
+        PipelineLauncher.JobState.DONE,
+        pipelineLauncher.getJobStatus(project, region, readInfo.jobId()));
+    double numRecords =
+        pipelineLauncher.getMetric(
+            project,
+            region,
+            readInfo.jobId(),
+            getBeamMetricsName(PipelineMetricsType.COUNTER, READ_ELEMENT_METRIC_NAME));
+    assertEquals(configuration.getNumRows(), numRecords, 0.5);
```

Comment:
Need publish metrics to BigQuery for both write and read pipeline, can refer to here: https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/6af4b6300edda4810cdf5108d8b1adff8984940a/it/google-cloud-platform/src/test/java/com/google/cloud/teleport/it/gcp/storage/FileBasedIOLT.java#L227

Reply:
Done